dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java

```diff
@@ -0,0 +1,572 @@
+        }
+
+        pconf.setDownloading(true);
+        pconf.setOperation(operation);
+
+        pconf.setAssets(assetsToPublish);
+        //Queries creation
+        pconf.setLuceneQueries(PublisherUtil.prepareQueries(tempBundleContents));
+        pconf.setId(bundleId);
+        pconf.setUser(APILocator.getUserAPI().getSystemUser());
+
+        //BUNDLERS
+
+        final List<Class<IBundler>> bundlers = new ArrayList<>();
+        final List<IBundler> confBundlers = new ArrayList<IBundler>();
+
+        final Publisher publisher = new PushPublisher();
+        publisher.init(pconf);
+        //Add the bundles for this publisher
+        for (final Class<IBundler> clazz : publisher.getBundlers()) {
+            if (!bundlers.contains(clazz)) {
+                bundlers.add(clazz);
+            }
+        }
+
+        final File bundleRoot = BundlerUtil.getBundleRoot(pconf);
+
+        // Run bundlers
+        BundlerUtil.writeBundleXML(pconf);
+        for (final Class<IBundler> aClass : bundlers) {
+
+            final IBundler bundler = aClass.newInstance();
+            confBundlers.add(bundler);
+            bundler.setConfig(pconf);
+            bundler.setPublisher(publisher);
+            final BundlerStatus bundlerStatus = new BundlerStatus(bundler.getClass().getName());
+            //Generate the bundler
+            Logger.info(DependencyBundlerTest.class, "Start of Bundler: " + aClass.getSimpleName());
+            bundler.generate(bundleRoot, bundlerStatus);
+            Logger.info(DependencyBundlerTest.class, "End of Bundler: " + aClass.getSimpleName());
+        }
+
+        pconf.setBundlers(confBundlers);
+
+        return pconf;
+    }
+
+    /**
+     * This test is for the filter excludeDependencyClasses.
+     *
+     * This test creates a Content Type and a content, and generates the bundle using the default Filter (this one does not have any exclude filter set),
+     * so the content and the content type (as a dependency) will be added.
+     * After this using the same Content Type and content generates another bundle but using a new created filter that exclude ContentTypes to be pushed
+     * as a dependency, so in the bundle the content will be added but the content type will not.
+     */
+    @Test
+    public void testGenerateBundleWithFilter_ExcludeDependencyClasses_ContentTypeWorkflow_NotAdded()
+            throws DotDataException, IllegalAccessException, DotBundleException, DotPublishingException, InstantiationException, DotPublisherException, IOException {
+        //Create ContentType
+        final ContentType contentType = TestDataUtils.getWikiLikeContentType();
+        //Create Content
+        final Contentlet content = TestDataUtils.getWikiContent(true,APILocator.getLanguageAPI().getDefaultLanguage().getId(),contentType.id());
+
+        //Create bundle with DefaultFilter
+        final Bundle bundleWithDefaultFilter = createBundle("TestBundle"+System.currentTimeMillis(),false,defaultFilterKey);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "TestBundle" appears 12 times in this file; the first occurrence is on line 216](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)